modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/processors/ExternalRefProcessor.java

```diff
@@ -92,6 +93,20 @@ public String processRefToExternalSchema(String $ref, RefFormat refFormat) {
             return renamedRef;
         }
 
+        RefFormat format = computeRefFormat($ref);
+        if (format.equals(RefFormat.RELATIVE)) {
+            String normalizedRef = "./" + Paths.get($ref).normalize().toString();
```

Comment:
To minimize impact on existing tests, this normalizes with a leading `./`.  It might be preferable to leave that out in normalized paths, but that would require changes to a number of tests that currently explicitly check for paths starting with `./`.

Reply:
It looks like this matches existing behavior so should be OK as-is; [the deserializer adds a leading `./` to relative references](https://github.com/swagger-api/swagger-parser/blob/153070c54acc6b96f902dd9e6155956921453847/modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java#L3935).

Given the existing behavior, an alternative approach to fixing this path resolution issue might be to normalize the relative path in `mungeRef` instead, although I'm not clear if that change would have to happen in only one or in both of `OpenAPIDeserializer.mungeRef` and `RefUtils.mungeRef`.

Reply:
After further investigation I'm fairly certain this is the only place we can effectively normalize ref paths.  The deserializer can't normalize nested refs and the resolver cache comes into play after we've already created duplicate models.